### PR TITLE
fix: windows hard coded well-known group name

### PR DESCRIPTION
**Pull request by tilo-chen**
**Issue #, if available:**

**Description of changes:**
`WindowsPlatform.java`: Lookup Windows group name "Everyone" and "Administrators" by SID instead of the hard-coded English string.

**Why is this change necessary:**
Non-English Windows versions can have localized name for groups such as "Everyone" and "Administrators". On those Windows versions, nucleus won't be able to install or function.

**How was this change tested:**
Manual test with German language Windows EC2 (AMI: `Windows_Server-2019-German-Full-Base-2021.09.15`)
Without this fix the test will fail:

```
[ERROR] Failures:
[ERROR]   WindowsPlatformTest.GIVEN_a_well_known_group_sid_WHEN_lookupGroupByIdentifier_THEN_succeed:292
Expected: "Everyone"
     but: was "Jeder"
```

**Any additional information or context required to review the change:**
To re-produce/test this, seems a non-English windows installation is required. If the system is English based, changing display language or system locale won't change the group names to localized ones.

**Checklist:**
 - [ ] Updated the README if applicable
 - [x] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 - [x] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java [hunk 1]
```diff
@@ -198,7 +203,7 @@ void GIVEN_file_system_permission_WHEN_convert_to_acl_THEN_succeed() throws IOEx
         int everyoneAclCount = 0;
         for (AclEntry aclEntry : initialOwnerAcl.getAcl()) {
             String name = aclEntry.principal().getName();
-            if (name.contains("Everyone")) {
+            if (name.contains(EVERYONE_GROUP_NAME)) {
```
Comment (rbattle) [inline src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java:206]: should this be `name.equals(EVERYONE_GROUP_NAME)`? could you create a group name like "Fake Everyone" and have this check pass?

File: src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java [hunk 2]
```diff
@@ -230,7 +235,7 @@ void GIVEN_file_system_permission_WHEN_convert_to_acl_THEN_succeed() throws IOEx
             everyoneAclCount = 0;
             for (AclEntry aclEntry : updatedAcl) {
                 String name = aclEntry.principal().getName();
-                if (name.contains("Everyone")) {
+                if (name.contains(EVERYONE_GROUP_NAME)) {
```
Comment (rbattle) [inline src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java:238]: same here

File: src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java (hunk 1 above)
Comment (MikeDombo) [reply src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java:206]: This is in our test code so it shouldn't really matter at all. But yeah, that is a reasonable change.

**Review by MikeDombo — APPROVED** [review]

File: src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java (hunk 1 above)
Comment (tilo-chen) [reply src/test/java/com/aws/greengrass/util/platforms/windows/WindowsPlatformTest.java]: Updated. Here `name` will actually equals "\Everyone"

**Review by junfuchen99 — APPROVED** [review]

**Comment by dpergarc** [comment]
Hi,

Would you mind sharing the .jar file so as it could be downloaded on the AWS console? 

Thanks, regards.
Daniel.

**Comment by tilo-chen** [comment]
> Hi,
> 
> Would you mind sharing the .jar file so as it could be downloaded on the AWS console?
> 
> Thanks, regards. Daniel.

We will release a version with this fix soon. Meanwhile, you may cherry-pick this change and build the jar with maven `package` command.

**Comment by dpergarc** [comment]
> > Hi,
> > Would you mind sharing the .jar file so as it could be downloaded on the AWS console?
> > Thanks, regards. Daniel.
> 
> We will release a version with this fix soon. Meanwhile, you may cherry-pick this change and build the jar with maven `package` command.

Hi tilo-chen,

I have built the source code and it appears to be working right now :)

Thanks,
Regards.

**Pull request merged**